Dz 23

belorusochkayul opened the pull request:
Привет!) Домашка по J.Unit.

belorusochkayul committed:
Add files with interfaces

belorusochkayul committed:
Delete AddBankName.java

belorusochkayul committed:
Delete AddMoney.java

belorusochkayul committed:
Delete AlfaBank.java

belorusochkayul committed:
Delete main.java

belorusochkayul committed:
Delete WithdrawMoney.java

belorusochkayul committed:
Delete Bank.java

belorusochkayul committed:
Delete AtmManufacturer.java

belorusochkayul committed:
Delete EnterSumReplenish.java

belorusochkayul committed:
Delete EnterSumWithdraw.java

belorusochkayul committed:
Add files via upload

upload files with edits

belorusochkayul committed:
add files without extends

belorusochkayul committed:
add the first task with ArrayList

belorusochkayul committed:
Add files with HashSet/HashMap

belorusochkayul committed:
Add files with Comporator

belorusochkayul committed:
Add the first task with calc

belorusochkayul committed:
Add task about files

belorusochkayul committed:
resolved conflicts in dz_9

belorusochkayul committed:
its all right

belorusochkayul committed:
Add files with J.Unit

tim-teacher commented on unit_анаграммы/AnnogrammaTest.java:
```diff
@@ -0,0 +1,20 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AnnogrammaTest {
```
Все ок, ничего переделывать не нужно, но что бы проверить больше вариантов и не писать много методов можно еще вот так сделать:
https://github.com/timofeykovalenko/coding-broadcast/blob/master/java/src/main/java/easy/anagram/AnagramTest.java
Просто, что бы знала что так можно.

tim-teacher commented on unit_комнаты/RoomTest.java:
```diff
@@ -0,0 +1,39 @@
+public class RoomTest {
+
+    @Test
+    public void testIsIlluminanceCorrect() {   //ошибка
```
Мне кажется метод isIlluminanceCorrect должен быть private и его не нужно тестить, а проверять нужно метод addLight, и проверить нужно на то что метод кидает исключения если попадаем в неверные условия, либо все ок если все верно.

А сейчас у тебя есть комната, затем ты создаешь  Light но нигде не используешь его, а затем проверяешь метод isIlluminanceCorrect и подаешь туда конкретное значение.

Дело в том что метод isIlluminanceCorrect это внутрення логика твоих классов, проверять тестами нужно то что публично, тоесть то что может кто то вызвать.

tim-teacher commented on unit_комнаты/RoomTest.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test
+    public void testIncorrectIsIlluminanceCorrect() {
```
Тоже самое тут.

tim-teacher commented on unit_комнаты/RoomTest.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test
+    public void testIsSpaceCorrect() {
```
Тоже самое, проверять нужно addSubject, а isSpaceCorrect это внутренняя реализация.

tim-teacher commented on unit_комнаты/RoomTest.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test
+    public void testIncorrectIsSpaceCorrect() {
```
Тоже самое.

tim-teacher commented on unit_регулярные/RegTest.java:
```diff
@@ -0,0 +1,21 @@
+    @Test
+    public void test() {
+        Reg regular = new Reg();
+        String result = regular.regCheck("\\d{4}\\s[A-Z]{2}[-][1-7]", "1234 PC-6");
```
Ой ой ой, а почему тест решает какую регулярку подать? Тоесть ты положила бизнес логику в тест. Тест ничего такого не должен знать. У тебя должен быть просто метод regCheck который принимает на вход строку и выдает результат, а регулярка должна быть внутри метода regCheck, а то как то странно выглядит, юнит тест сам создал логику и сам проверил. Тоесть мы тут ничего по факту не проверяем. Я могу спокойно использовать эту функцию и подать туда неправильную регулярку и тест мне ничего не покажет.

PS: метод regCheck нужно переименовать в checkCurNumber. Метод должен делать конкретную вешь.

belorusochkayul committed:
Add files via upload

Привет! Со всеми прошедшими праздниками!)
Тест для регулярок....

belorusochkayul committed:
Add files via upload

Привет! Все тесты в одном классе "RoomTest", потому что методы все в Room собраны...
P.S.  Не смогла проверить исключения на "неправильность". Нашла, что должен быть метод "assertDoesNotThrow​()", в 5-том JU должен быть, но он недоступен почему то  у меня...

tim-teacher commented on Reg.java:
```diff
@@ -0,0 +1,31 @@
+public class Reg {
+
+    final static String BY = "BY";
+    final static String NOTBY = "NOT BY";
```
private

tim-teacher commented on Reg.java:
```diff
@@ -0,0 +1,31 @@
+        return formResult;
+    }
+
+    public boolean regCarNumber(String reg, String userInputNumber) {
```
rename to isBelarusCarNumber

и убрать String reg переменную, а userInputNumber переименовать в carNumber.

tim-teacher commented on Reg.java:
```diff
@@ -0,0 +1,31 @@
+    final static String BY = "BY";
+    final static String NOTBY = "NOT BY";
+
+    public String userInputNumberFormatted(String userInputNumber) {
```
переименовать, в что то понятное.

tim-teacher commented on Reg.java:
```diff
@@ -0,0 +1,31 @@
+    public boolean regCarNumber(String reg, String userInputNumber) {
+        Pattern pattern = Pattern.compile(reg);
+        Matcher matcher = pattern.matcher(userInputNumber);
+        if (matcher.matches()) {
```
можно просто поменять на return matcher.matches();

tim-teacher commented on Reg.java:
```diff
@@ -0,0 +1,31 @@
+        }
+    }
+
+    public String IsCarNumberIsBelorussianPrint(String userInputNumber, boolean result) {
```
public String formatCarNumbe(String carNumber, boolean isBalarusCarNumber) {
        if (isBalarusCarNumber) {
            return carNumber + BY;
        } else { 
            return carNumber + NOTBY;
        }
    }

tim-teacher commented on Reg.java:
```diff
@@ -0,0 +1,31 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Reg {
```
Можно все методы сделать статичными и вызывать из через Reg.method

тоесть не нужно будет создавать объект.

tim-teacher replied:
и переименовать.

tim-teacher commented on RegTest.java:
```diff
@@ -0,0 +1,35 @@
+public class RegTest {
+
+    @Test
+    public void IsCarNumberIsBelorussianPrintBY() {
```
@Test
    public void formatShouldAddNumbeWithBySuffix() {
        String carNumber = "1564 MP-7";
        String expected = carNumber + " BY";
        boolean isBelarusCarNumber = true;

        String actual = Reg.formatCarNumber(carNumber, isBelarusCarNumber);
        Assert.assertEquals(expected, actual);
    }

tim-teacher commented on RegTest.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @Test
+    public void IsCarNumberIsBelorussianPrintNotBY() {
```
@Test
    public void formatShouldAddNumbeWithNoBySuffix() {
         String carNumber = "MP 743 5";
        String expected = carNumber + " NOT BY";
        boolean isBelarusCarNumber = false;

        String actual = Reg.formatCarNumber(carNumber, isBelarusCarNumber);
        Assert.assertEquals(expected, actual);
    }

tim-teacher commented on RegTest.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @Test
+    public void IsCarNumberIsBelorussianPrintIncorrect1() {
```
remove

tim-teacher commented on RegTest.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @Test
+    public void IsCarNumberIsBelorussianPrintIncorrect2() {
```
remove

tim-teacher commented on unit_анаграммы/Annogramma.java:
```diff
@@ -0,0 +1,25 @@
+
+public class Annogramma {
+
+    public boolean checkAnagram(String firstWord, String secondWord) {
```
Можно сделать static.

tim-teacher commented on unit_комнаты/RoomTest.java:
```diff
@@ -0,0 +1,78 @@
+public class RoomTest {
+
+    @Test
+    public void IsIlluminanceLowLimitCorrect() {
```
Названия методов с маленькой буквы

tim-teacher commented on unit_комнаты/RoomTest.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Test
+    public void IncorrectIsIlluminanceLowLimitCorrect() {
```
@Test
    public void IsIlluminanceLowLimitCorrect() {
        int roomSquare = 300;
        int windowQuantity = 0;

        Room room = new Room("стол", roomSquare, windowQuantity);
        boolean result = room.isIlluminanceLowLimitCorrect(301);
        Assert.assertTrue(result);
    }

tim-teacher replied:
и в остальных тоже так, имею ввиду вынести магические числа в переменные.

belorusochkayul committed:
Add files with JUnit task

belorusochkayul committed:
Delete Main.java

belorusochkayul committed:
Delete RegTest.java

belorusochkayul committed:
Delete Reg.java

belorusochkayul committed:
Delete Main.java

belorusochkayul committed:
Delete Reg.java

belorusochkayul committed:
Delete RegTest.java

belorusochkayul committed:
Delete Annogramma.java

belorusochkayul committed:
Delete AnnogrammaTest.java

belorusochkayul committed:
Delete MainUnits.java

belorusochkayul committed:
Delete Light.java

belorusochkayul committed:
Delete Building.java

belorusochkayul committed:
Delete Subject.java

belorusochkayul committed:
Delete SpaceUsageTooMuchException.java

belorusochkayul committed:
Delete RoomTest.java

belorusochkayul committed:
Delete Room.java

belorusochkayul committed:
Delete Main.java

belorusochkayul committed:
Delete IlluminanceTooMuchException.java

belorusochkayul committed:
Delete IlluminanceTooLowException.java

belorusochkayul committed:
Delete Describe.java

belorusochkayul committed:
JU

belorusochkayul commented:
Привет! Исправленные файлы загружены под комментарием "Add files with JUnit task"
